### PR TITLE
Add Github action to build project

### PR DESCRIPTION
**Pull request by timothy-ch-cheung**
Add github action workflow:
builds project + run tests + coverage to codecov

Add unit tests to TabBar and TopBar

### CODE REVIEW

File: src/components/TopBar.tsx [hunk 1]
```diff
@@ -69,8 +69,8 @@ export default function TopBar(props: Props): JSX.Element {
     <Container>
       <Content>
         {backButton && (
-          <BackIconLink onClick={() => Router.back()}>
-            <BackIcon />
+          <BackIconLink onClick={() => Router.back()} data-test="top-bar-back-icon">
```
Comment (pavsidhu) [inline src/components/TopBar.tsx:72]: What does this do?

File: test/__mocks__/ComponentStub.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React, { ReactElement } from "react"
+
+const ComponentStub = (): ReactElement => {
+  return <></>
+}
+
+export default ComponentStub
```
Comment (pavsidhu) [inline test/__mocks__/ComponentStub.tsx:7]: Missing EOF

File: test/__mocks__/fileMock.js [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = "test-file-stub"
```
Comment (pavsidhu) [inline test/__mocks__/fileMock.js:1]: Missing EOF here too

**Review by pavsidhu — CHANGES_REQUESTED** [review]

File: src/components/TopBar.tsx (hunk 1 above)
Comment (timothy-ch-cheung) [reply src/components/TopBar.tsx:72]: It's used to select the element for testing. The purpose is to allow the element to be selected independent of style and nesting changes. (https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements)

File: test/__mocks__/ComponentStub.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React, { ReactElement } from "react"
+
+const ComponentStub = (): ReactElement => {
```
Comment (adamjhc) [inline test/__mocks__/ComponentStub.tsx:3]: can the `export default` be inline here, please

**Review by adamjhc — CHANGES_REQUESTED** [review]

File: test/__mocks__/ComponentStub.tsx [hunk 5]
```diff
@@ -1,7 +1,5 @@
 import React, { ReactElement } from "react"
 
-const ComponentStub = (): ReactElement => {
+export const ComponentStub = (): ReactElement => {
```
Comment (adamjhc) [inline test/__mocks__/ComponentStub.tsx:3]: Any reason for no `default`?

```suggestion
export default const ComponentStub = (): ReactElement => {
```

If you add `default` then you don't need to add the `{}` around it when importing

Comment (timothy-ch-cheung) [reply test/__mocks__/ComponentStub.tsx:3]: I don't think that's valid syntax:
image

Also I read this which gave some good reasons against default exports:
https://basarat.gitbook.io/typescript/main-1/defaultisbad

Comment (adamjhc) [reply test/__mocks__/ComponentStub.tsx:3]: mb it works with 
```typescript
export default function ComponentStub(): ReactElement {}
```

Those seem good reasons for not default exporting but for a file like this which seems to me is never going to change and is not part of a library, `default` makes things cleaner. @pavsidhu you have an opinion on this?

Comment (pavsidhu) [reply test/__mocks__/ComponentStub.tsx:3]: Yeahh I agree with @adamjhc on this, keep main components as default export the same way as in the example above. Also yeah default export is bad but it's generally standard to export components as the default export, we could ignore this but then libraries we'd have to use default imports too so it doesn't really make sense

**Review by adamjhc — APPROVED** [review]

**Review by pavsidhu — APPROVED** [review]

**Pull request merged**